### DISCUSSION TITLE
Memory leak when using validator in long-running tasks

### DISCUSSION THREAD

**Discussion by DavideTriso** [symfony/symfony#45413] [Q&A] [answered]
In a project I am working on I have a background long-running task (scheduled with a crontab) which syncs our data with third-party softwares through different web APIs. This task makes use of the validator component to validate the data we send to and receive from the third-party softwares.

I have noticed that the memory consumption increases after each data validation the task performs and is never released, causing memory leaks. It seems like the validator component holds all the validation results in memory (memoization).

Does anyone know if the validator components does actually memoize the results and how to overcome this problem in long-running tasks?

Tanks!

**Comment by ghost** [comment]
Can you make a reproducer app? If it is actually a Validator causing the problem then it should be reproducible on any app.

**Reply by DavideTriso** [reply]
@javaDeveloperKid thank you for your quick reply. I will try to provide an example app as soon as possible.

**Comment by nuryagdym** [comment, marked as answer]
After checking the validator code I found the issue 
```php
for ($i = 0; $i < 100000; $i++) {
    $this->validator->validate('aa', [new \Symfony\Component\Validator\Constraints\NotBlank()]);
    $this->validator->reset(); //this solves memory leak issue;
}
```

However, in production env, it will be validator logger than in dev environment which does not have `reset()` method.
